### DISCUSSION TITLE
Arm planning in the ROS API.

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#65] [Q&A]
I noticed that in the Python API, we have functions like move_end_effector_pose, where we can specify a point in space and have the end effector move there. However, there does not seem to be a similar function in the ROS API, which we have to use due to certain requirements in our project. May I please know where I can fund functions and scripts that do this planning? Thank you very much!

**Comment by masudam** [comment]
The Python API (ihsrb) is provided as a Python package, and it is not specifically encrypted, allowing you to view the code. You can refer to it to use it as a ROS API.

You can find the location of the files included in the package like this: `dpkg -L ros-noetic-hsrb-interface-py.`

However, please remember that the code is copyrighted by TMC, so there may be restrictions on its public disclosure.
